plugins/woocommerce/src/Internal/Fulfillments/FulfillmentsRenderer.php

```diff
@@ -227,14 +230,56 @@ public function render_fulfillment_drawer_slot() {
 	/**
 	 * Loads the payment method promotions scripts and styles.
 	 */
-	public static function load_components() {
-		if ( ! self::should_render_fulfillment_drawer() ) {
+	public function load_components() {
+		if ( ! $this->should_render_fulfillment_drawer() ) {
 			return;
 		}
 		WCAdminAssets::register_style( 'fulfillments', 'style', array( 'wp-components' ) );
 		WCAdminAssets::register_script( 'wp-admin-scripts', 'fulfillments', true );
 	}
 
+	/**
+	 * Prints the fulfillments object in the admin header.
+	 */
+	public function print_fulfillments_object() {
```

Comment:
How come [wp_localize_script](https://developer.wordpress.org/reference/functions/wp_localize_script/) is not used instead?

Reply:
It needs a registered script which I don't have :)

Reply:
Okay, I managed to convert it to `wp_localize_script`. I just needed to prepend `wc-admin-` to the handle I provided. The replacement is in [71ea793](https://github.com/woocommerce/woocommerce/pull/58731/commits/71ea793ae5fcd44e1217f5753a3c1a3f90c61387)[71ea793](https://github.com/woocommerce/woocommerce/pull/58731/commits/71ea793ae5fcd44e1217f5753a3c1a3f90c61387).

Reply:
Okay, needed to revert it because it needs a registered script, and we don't register the fulfillment drawer script on the customer area, but we render the settings object. And I can't register a script without a source file, so it's better to keep it like that.

Reply:
Created follow up issue: [WOOPLUG-4693](https://linear.app/a8c/issue/WOOPLUG-4693/order-fulfillments-clean-up-script-loading)